system/lib/libc/musl/src/thread/__timedwait.c

```diff
@@ -42,7 +42,10 @@ int __timedwait_cp(volatile int *addr, int val,
 #ifdef __EMSCRIPTEN__
 	double msecsToSleep = top ? (top->tv_sec * 1000 + top->tv_nsec / 1000000.0) : INFINITY;
 	int is_main_thread = emscripten_is_main_browser_thread();
-	if (is_main_thread || pthread_self()->cancelasync == PTHREAD_CANCEL_ASYNCHRONOUS) {
+	int can_cancel_wait = pthread_self()->canceldisable == PTHREAD_CANCEL_ENABLE ||
+		(pthread_self()->canceldisable == PTHREAD_CANCEL_MASKED &&
+		 pthread_self()->cancelasync == PTHREAD_CANCEL_ASYNCHRONOUS);
```

Comment:
Let's store `pthread_self()` to its own variable, there is no guarantee that it would be inlined.